runtime/coverage.go

```diff
@@ -483,3 +484,42 @@ func (r *CoverageReport) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+// MarshalLCOV serializes each common.Location/*LocationCoverage
+// key/value pair on the *CoverageReport.Coverage map, to the
+// LCOV format. Currently supports only line coverage, function
+// and branch coverage are not yet available.
+// Description for the LCOV file format, can be found here
+// https://github.com/linux-test-project/lcov/blob/master/man/geninfo.1#L948.
+func (r *CoverageReport) MarshalLCOV() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	for location, coverage := range r.Coverage { // nolint:maprange
+		_, err := fmt.Fprintf(buf, "TN:\nSF:%s\n", location.ID())
+		if err != nil {
+			return nil, err
+		}
+		lines := make([]int, 0)
```

Comment:
I think you can allocate directly lines of size len(coverage.LineHits) to avoid memory reallocations later.

Reply:
Good catch, updated in https://github.com/onflow/cadence/pull/2577/commits/f5f4602fd9dfde924aac7eb1151abcf8979d3356